modules/web/js/ballerina/diagram2/views/default/components/nodes/resource-node.jsx

```diff
@@ -107,9 +107,19 @@ class ResourceNode extends React.Component {
         tLinkBox.y += annotationBodyHeight;
         const thisNodeIndex = parentNode.getIndexOfResources(this.props.model);
         const resourceSiblings = parentNode.getResources();
-        let showAddResourceBtn = true;
-        if (parentNode.getProtocolPackageIdentifier().value === 'ws' && resourceSiblings.length >= 6) {
-            showAddResourceBtn = false;
+        // For Web sockets
+        let showAddResourceBtnForWS = true;
+        if (parentNode.getProtocolPackageIdentifier().value === 'ws' &&
+            resourceSiblings.length >= 6) {
```

Comment:
Cant we bring this to line 112 ? To be on the earlier line .

Reply:
Fixed with 7907698b7efdcf1142c2ba9a166ea8e0ff63c5b0